src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IDelegateCreationExpression.cs

```diff
@@ -0,0 +1,1418 @@
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void DelegateCreationExpression_ImplicitMethodBinding_MultipleCandidates()
+        {
+            string source = @"
+using System;
+class Program
+{
+    void Main()
+    {
+        /*<bind>*/Action<int> a = M1;/*</bind>*/
+    }
+    void M1(object o) { }
+    void M1(int i) { }
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'Action<int> a = M1;')
+  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'Action<int> a = M1;')
+    Variables: Local_1: System.Action<System.Int32> a
+    Initializer: IDelegateCreationExpression (OperationKind.DelegateCreationExpression, Type: System.Action<System.Int32>) (Syntax: 'M1')
+        Target: IMethodBindingExpression: void Program.M1(System.Int32 i) (OperationKind.MethodBindingExpression, Type: null) (Syntax: 'M1')
+            Instance Receiver: IInstanceReferenceExpression (OperationKind.InstanceReferenceExpression, Type: Program) (Syntax: 'M1')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<LocalDeclarationStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void DelegateCreationExpression_ExplicitDelegateConstructorImplicitMethodBinding_MultipleCandidates()
+        {
+            string source = @"
+using System;
+class Program
+{
+    void Main()
+    {
+        Action<string> a = /*<bind>*/new Action<string>(M1)/*</bind>*/;
+    }
+
+    void M1(object o) { }
+
+    void M1(string s) { }
+}
+";
+            string expectedOperationTree = @"
+IDelegateCreationExpression (OperationKind.DelegateCreationExpression, Type: System.Action<System.String>) (Syntax: 'new Action<string>(M1)')
+  Target: IOperation:  (OperationKind.None) (Syntax: 'M1')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<ObjectCreationExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void DelegateCreationExpression_ExplicitDelegateConstructorImplicitMethodBinding_MultipleCandidates_InvalidNoMatch()
+        {
+            string source = @"
+using System;
+class Program
+{
+    void Main()
+    {
+        Action<int> a = /*<bind>*/new Action<int>(M1)/*</bind>*/;
+    }
+
+    void M1(object o) { }
+
+    void M1(string s) { }
+}
+";
+            string expectedOperationTree = @"
+IInvalidExpression (OperationKind.InvalidExpression, Type: System.Action<System.Int32>, IsInvalid) (Syntax: 'new Action<int>(M1)')
```

Comment:
The result looks inconsistent with DelegateCreationExpression_ImplicitMethodBinding_MultipleCandidates_InvalidNoMatch test.

Reply:
Similar to your comment above, we'll have to make changes to the binder to fix this.

---
In reply to: [140107720](https://github.com/dotnet/roslyn/pull/22056#discussion_r140107720) [](ancestors = 140107720)